cors/preflight/handler.go

```diff
@@ -0,0 +1,77 @@
+package preflight
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+type Methods []string
+
+func handler(allowedMethods, allowedHeaders []string) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if allowedMethods := strings.Join(allowedMethods, ", "); allowedMethods != "" {
+			w.Header().Set("Access-Control-Allow-Methods", allowedMethods)
+		}
+
+		if allowedHeaders := strings.Join(allowedHeaders, ", "); allowedHeaders != "" {
+			w.Header().Set("Access-Control-Allow-Headers", allowedHeaders)
+		}
+	}
+}
+
+func AddHandler(router *mux.Router, allowedHeaders ...string) {
```

Comment:
Maybe we can leave this to the consuming API?

Because you should be able to differ the headers list per endpoint?

Reply:
I just thought would be simplest for the API consumers, and I don't really see the problem with allowing some headers on endpoints that don't use them.

If we are going to allow different headers on each endpoint I guess we would need to make a `Preflight` struct, with methods like `AddAllowedHeaders(path string, headers []string)` or something similar? Or did you have another solution in mind?